src/daemon/daemon.py

```diff
@@ -0,0 +1,160 @@
+from db import create_address, db_host, db_name, get_highest_block_number_in_db, save_address, save_block
+from mongoengine import connect, disconnect
+from rpc import get_block, get_block_count
+import argparse
+import logging
+import sys, os, time, atexit
+from _signal import SIGTERM
+
+class Daemon:
+    def __init__(self, pidfile, interval, capacity):
+        self.pidfile = pidfile
+        self.interval = interval
+        self.capacity = capacity
+        self.pids = []
+        
+    def _stall_if_running_by_docker(self, pid):
+        if self.capacity == 'docker':
+            while True:
+                try:
+                    os.kill(pid, 0)
+                except OSError:
+                    break
+        sys.exit(0)
+        
+    def daemonize(self):
```

Comment:
Daemonization seems to be unnecessary when running inside a docker container

Reply:
Removed demonization.